src/base/searchengine.h

```diff
@@ -38,10 +40,119 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
```

Comment:
If you did this "general" Version template, then put it in its own header. Otherwise, what sense in it?